### PR TITLE
Added 'segal-decorators'

### PR DESCRIPTION
**Pull request by danrevah**
**By submitting this pull request, I promise I have read the contribution guidelines twice and ensured my submission follows it. I realize not doing so wastes the maintainers' time that they could have spent making the world better. 🖖**

⬆⬆⬆⬆⬆⬆⬆⬆⬆⬆

I haven't encountered any similar library in `awesome-nodejs`, it brings a new elegant approach of commonly used functions (**memoize, debounce, throttle**), while also bringing some new useful functionality such as **timeMemoize, lruMemoize, asyncRetry**, which could be useful in writing more elegant, clean & useful code.

Link: https://github.com/danrevah/segal-decorators

### CODE REVIEW

**Comment by FarazPatankar** [comment]
Hey @danrevah,

Have you read the contributing guidelines for the repository? I noticed that your PR does not adhere to all the mentioned guidelines.

Please read them and update your PR. Drop a comment here when you do and we will take another look.

**Comment by danrevah** [comment]
@FarazPatankar Thanks. I have read the contributing guidelines.

Let me know if I miss something, but seems like I just didn't follow this guideline: "The pull request should have a useful title and include a link to the package and why it should be included."

Updating..

**Comment by danrevah** [comment]
@FarazPatankar Done. 

Also updated commit message.

File: readme.md [hunk 1]
```diff
@@ -183,7 +183,7 @@
 - [RxJS](https://github.com/reactivex/rxjs) - Functional reactive library for transforming, composing, and querying various kinds of data.
 - [Lazy.js](https://github.com/dtao/lazy.js) - Utility library similar to lodash/Underscore but with lazy evaluation, which can translate to superior performance in many cases.
 - [Kefir.js](https://github.com/kefirjs/kefir) - Reactive library with focus on high performance and low memory usage.
-
+- [Segal Decorators](https://github.com/danrevah/segal-decorators) - Bunch of highly useful decorators, helps in writing a more concise code while improving readability.
```
Comment (FarazPatankar) [inline readme.md]: I am just suggesting a change that I feel adheres better to these two requirements:
- Keep descriptions short and simple, but descriptive.
- Check your spelling and grammar.

Perhaps you could change the description to:
`Collection of highly useful decorators that help write concise code while also improving readability.`

Comment (danrevah) [reply readme.md]: Ack.

Changed.

**Comment by danrevah** [comment]
@FarazPatankar Fair enough, fixed it.

**Comment by sindresorhus** [comment]
JavaScript doesn't yet have decorators, so I'm assuming your library is for TypeScript and Babel users, although you don't make that clear (which you should). I think this would be a better fit for https://github.com/dzharii/awesome-typescript and https://github.com/sorrycc/awesome-javascript

**Pull request closed without merging**